src/mpid/ch4/include/mpidpre.h

```diff
@@ -227,6 +227,17 @@ typedef struct MPIDI_prequest {
     MPI_Datatype datatype;
 } MPIDI_prequest_t;
 
+typedef struct MPIDI_part_request {
+    /* partitioned attributes */
+    void *buffer;
+    int partitions;
+    MPI_Aint count;             /* count per partition */
+    int rank;
+    int tag;
+    MPIR_Context_id_t context_id;
```

Comment:
The `comm` should be stored in the request already, right? (If not, it should). Why do we need to store `context_id` here? It is redundant.

Reply:
You are right that we don't need `context_id` for request created by `psend_init|precv_init`, since `comm` is already in `MPIR_Request`. But we need it for unexpected rreq which is temporarily created when the receiver receives the message matching AM (see  `MPIDIG_part_send_init_target_msg_cb`)

Reply:
> But we need it for unexpected rreq which is temporarily created when the receiver receives the message matching AM (see `MPIDIG_part_send_init_target_msg_cb`)

I see. We look up the `root_comm` when we receive the unexpected am message, right? So you can just set the comm to `root_comm`, right?

Reply:
Reverted the change, we need `context_id` set for both posted_rreq and unexp_rreq so that both types can be matched with the same routine `MPIDIG_part_dequeue`.

I don't use `root_comm` in partitioned pt2pt am matching. They are separate queues from the regular pt2pt queues anyway. I feel that involving `root_comm` just makes the logic complex unnecessarily. Simply compare context_id is more easy to understand.